python/cuxfilter/charts/core/non_aggregate/core_graph.py

```diff
@@ -254,19 +256,22 @@ def box_callback(xmin, xmax, ymin, ymax):
             )
 
             dashboard_cls._query_str_dict[self.name] = query
-            temp_data = dashboard_cls._query(
-                dashboard_cls._generate_query_str()
-            )
-            node_ids = temp_data[self.node_id]
+            nodes = dashboard_cls._query(dashboard_cls._generate_query_str())
 
-            nodes, edges = self.query_graph(node_ids, self.nodes, self.edges)
+            if self.inspect_neighbors._active:
+                node_ids = nodes[self.node_id]
+                nodes, edges = self.query_graph(
+                    node_ids, self.nodes, self.edges
+                )
+            else:
+                edges = None
```

Comment:
Purely a personal style pref but I think it would be slightly clearer as
```
# set both nodes and edges to initial values together in one place
nodes = dashboard_cls._query(dashboard_cls._generate_query_str())
edges = None

# conditionally update both nodes and edges only in one place (no else: needed)
if self.inspect_neighbors._active:
    node_ids = nodes[self.node_id]
    nodes, edges = self.query_graph(node_ids, self.nodes, self.edges)
```

Reply:
fixed, thanks